src/common/util.ts

```diff
@@ -50,6 +50,7 @@ export const arrContains = <T>(arr: T[], val: T): boolean => {
 }
 
 export const asyncWaterfall = async <T>(val: T, operations: ((val: T) => Promise<T>)[]): Promise<T> => {
+  // Why not reduce?
```

Comment:
@dholms why the manual iteration here?

Reply:
the alternative is an `await/async` reduce. Totally possible, but I thought that this was clearer & also apparently more performant (https://stackoverflow.com/a/41243567)

Reply:
it'd have to be `await/async` reduce which is possible, but I thought that this was clearer (and is apparently more performant: https://stackoverflow.com/a/49198393)

Reply:
> this was clearer

Interesting. I find `reduce` to be much more descriptive of what's happening, since it's a more structured abstraction. `for` loops admit all kinds of control flow trickery not allowed in a `reduce`. To each their own 🤷‍♀️

> apparently more performant

I did some further digging. The performance improvement is from not needing to call `await` on the accumulator (because it's an argument marked by `async`). This should be negligible because the network far outstrips the `await` resolution. Another fascinating edge case in the JS semantics!

Reply:
Yeah I generally agree on `reduce` being more descriptive & like it better. But with await/async, the accumulator being a promise & this for loop being super self contained, this just seemed like the less headachey way to do it